control-plane-operator/controllers/hostedcontrolplane/csi/kubevirt/kubevirt.go

```diff
@@ -0,0 +1,397 @@
+		panic(err)
+	}
+
+	return obj
+}
+
+func mustClusterRole(file string) *rbacv1.ClusterRole {
+	b := getContentsOrDie(file)
+	obj := &rbacv1.ClusterRole{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(b), 500).Decode(&obj); err != nil {
+		panic(err)
+	}
+
+	return obj
+}
+
+func mustClusterRoleBinding(file string) *rbacv1.ClusterRoleBinding {
+	b := getContentsOrDie(file)
+	obj := &rbacv1.ClusterRoleBinding{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(b), 500).Decode(&obj); err != nil {
+		panic(err)
+	}
+
+	return obj
+}
+
+func mustRole(file string) *rbacv1.Role {
+	b := getContentsOrDie(file)
+	obj := &rbacv1.Role{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(b), 500).Decode(&obj); err != nil {
+		panic(err)
+	}
+
+	return obj
+}
+
+func mustRoleBinding(file string) *rbacv1.RoleBinding {
+	b := getContentsOrDie(file)
+	obj := &rbacv1.RoleBinding{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(b), 500).Decode(&obj); err != nil {
+		panic(err)
+	}
+
+	return obj
+}
+
+func getContentsOrDie(file string) []byte {
+	f, err := resources.Open("files/" + file)
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			panic(err)
+		}
+	}()
+	b, err := io.ReadAll(f)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+func reconcileInfraConfigMap(cm *corev1.ConfigMap) error {
+	cm.Data = map[string]string{
+		"infraClusterNamespace": cm.Namespace,
+		"infraClusterLabels":    "",
+	}
+	return nil
+}
+
+func reconcileController(controller *appsv1.Deployment, componentImages map[string]string, deploymentConfig *config.DeploymentConfig) error {
+	controller.Spec = *controllerDeployment.Spec.DeepCopy()
+
+	csiDriverImage, exists := componentImages["kubevirt-csi-driver"]
+	if !exists {
+		return fmt.Errorf("unable to detect kubevirt-csi-driver image from release payload")
+	}
+
+	csiProvisionerImage, exists := componentImages["csi-external-provisioner"]
+	if !exists {
+		return fmt.Errorf("unable to detect csi-external-provisioner image from release payload")
+	}
+
+	csiAttacherImage, exists := componentImages["csi-external-attacher"]
+	if !exists {
+		return fmt.Errorf("unable to detect csi-external-attacher image from release payload")
+	}
+
+	csiLivenessProbeImage, exists := componentImages["csi-livenessprobe"]
+	if !exists {
+		return fmt.Errorf("unable to detect csi-livenessprobe image from release payload")
+	}
+
+	for i, container := range controller.Spec.Template.Spec.Containers {
+		switch container.Name {
+		case "csi-driver":
+			controller.Spec.Template.Spec.Containers[i].Image = csiDriverImage
+		case "csi-provisioner":
+			controller.Spec.Template.Spec.Containers[i].Image = csiProvisionerImage
+		case "csi-attacher":
+			controller.Spec.Template.Spec.Containers[i].Image = csiAttacherImage
+		case "csi-liveness-probe":
+			controller.Spec.Template.Spec.Containers[i].Image = csiLivenessProbeImage
+		}
```

Comment:
@enxebre the images in the yaml are overwritten using images from the release payload here.